[MusicXML] Properly export Persian key accidentals

rettinghaus opened the pull request:
This PR adds the missing `key-alter` values for Persian [sori](https://en.wikipedia.org/wiki/Sori_(music)) and [koron](https://en.wikipedia.org/wiki/Koron_(music)) key accidentals to MusicXML export.

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [x] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [ ] I created a unit test or vtest to verify the changes I made (if applicable)

rettinghaus commented:
Small test file with sori and koron as key accidentals: [SoriKoronKey.mscz.zip](https://github.com/musescore/MuseScore/files/12021657/SoriKoronKey.mscz.zip)

cbjeukendrup commented:
It would be great if you could add this test file as a unit test. See `musicxml_tests.cpp`; `mxmlMscxExportTestRef` might be the most appropriate for this.

Jojo-Schmitz commented on src/importexport/musicxml/internal/musicxml/exportxml.cpp:
```diff
@@ -2136,6 +2136,10 @@ static double accSymId2alter(SymId id)
         break;
     case SymId::accidentalDoubleSharp:                     res =  2;
         break;
+    case SymId::accidentalSori:                            res =  0.5;
```
In accidental.cpp the offsets are 0.33 and -0.67, respectively, changed to that via 56e76ba after I found some evidence for that, see https://de.wikipedia.org/wiki/Dastgah#Die_Dastg%C4%81hs_mit_ihren_%C4%80w%C4%81z:

**Erläuterung: B=H; is: um einen Halbton erhöht; b: um einen Halbton erniedrigt; p [„koron“]: um grob einen Viertelton (etwas genauer: um 2/3 eines Halbtons)[[30]](https://de.wikipedia.org/wiki/Dastgah#cite_note-30) erniedrigt; > [„sori“]: um grob einen Viertelton (genauer: um 1/3 eines Halbtons) erhöht.**

rettinghaus replied:
Agreed.

rettinghaus commented:
Updated the PR to also include correct values for arrow quarter-tone accidentals.

@cbjeukendrup I saw that there is no test for non-standard key signatures in general. I'll add one in a separate PR.

Jojo-Schmitz commented:
I was about to mention those other accitentals, you beat me to it ;-)

Jojo-Schmitz commented:
There are quite a few more in accidentals.cpp and also in musicxmlsupport.cpp, at least the latter should get added here to IMHO

rettinghaus commented:
@Jojo-Schmitz I took the list from `accidentals.cpp` and adjusted the values for export to `key-alter` in MusicXML. Found on the way a small issue there and fixed it. Please have another look.

Jojo-Schmitz commented on src/importexport/musicxml/internal/musicxml/exportxml.cpp:
```diff
@@ -2306,8 +2306,6 @@ static double accSymId2alter(SymId id)
     case SymId::accidental4CommaFlat:                      res = -0.89;
         break;
-    case SymId::accidentalSharp:                           res =  0.89;
-        break;
```
Maybe leave that in but commented out, like in accidental.cpp?

Jojo-Schmitz replied:
Except for consistency. Maybe also leave a comment in this file and in accidentals.cpp, reg. the need to keep them in sync

Jojo-Schmitz commented on src/importexport/musicxml/tests/data/testNonStandardKeySig3.xml:
```diff
@@ -7,8 +7,8 @@
     <encoding>
-      <software>MuseScore 0.7.0</software>
-      <encoding-date>2007-09-10</encoding-date>
+      <software>MuseScore 4.2.0</software>
```
I believe this change isn't needed/wanted

Jojo-Schmitz commented on src/importexport/musicxml/tests/data/testNonStandardKeySig3.xml:
```diff
@@ -22,6 +22,7 @@
       <part-abbreviation>Pno.</part-abbreviation>
       <score-instrument id="P1-I1">
         <instrument-name>Piano</instrument-name>
+        <instrument-sound>keyboard.piano</instrument-sound>
```
and this a follow up, also not needed

Jojo-Schmitz replied:
Seems I was wrong on that, sorry about this.

rettinghaus replied:
Well, the test are green again …

Jojo-Schmitz commented:
Not all

cbjeukendrup commented:
Is this ready or does it still need discussion or changes?

Jojo-Schmitz commented:
I believe at least those Extended Helmholtz-Ellis accidentals don't have a 0 offset should be part of this PR

rettinghaus commented:
This is ready, @cbjeukendrup.  
I will not add special treatment for Extended Helmholtz-Ellis accidentals in this PR. I don't see the point in having support for only a fraction of a range of barely used accidentals here.
They can be added when all correct values have been added to `accidental.cpp`.

Jojo-Schmitz commented:
OK, fine by me.
Do you happen to know what the correct values might be or how to find out?

rettinghaus commented:
Actually no, otherwise I would have just added them.

sammik commented:
> Extended Helmholtz-Ellis accidentals

see
https://en.xen.wiki/w/Helmholtz-Ellis_notation
https://marsbat.space/pdfs/HEJI2_legend+series.pdf

rettinghaus commented:
Thanks @sammik. I propose to add these in a separate PR.

Jojo-Schmitz commented:
Agreed
